tools/pd-analysis/analysis/parse_log_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analysis
+
+import (
+	. "github.com/pingcap/check"
+	"testing"
+)
+
+func TestParser(t *testing.T) {
+	TestingT(t)
+}
+
+var _ = Suite(&testParseLog{})
+
+type testParseLog struct{}
+
+func utilTransferRegionCountParseLog(operator, content string, expect []uint64) bool {
+	r := GetTransferRegionCounter().CompileRegex(operator)
+	results := GetTransferRegionCounter().parseLine(content, r)
+	if len(results) != len(expect) {
+		return false
+	}
+	for i := 0; i < len(results); i++ {
+		if results[i] != expect[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func (t *testParseLog) TestTransferRegionCountParseLog(c *C) {
+	{
+		operator := "balance-leader"
+		content := "[2019/09/05 04:15:52.404 +00:00] [INFO] [operator_controller.go:119] [\"operator finish\"] [region-id=54252] [operator=\"\"balance-leader {transfer leader: store 4 to 6} (kind:leader,balance, region:54252(8243,398), createAt:2019-09-05 04:15:52.400290023 +0000 UTC m=+91268.739649520, startAt:2019-09-05 04:15:52.400489629 +0000 UTC m=+91268.739849120, currentStep:1, steps:[transfer leader from store 4 to store 6]) finished\"\"]"
+		var expect = []uint64{54252, 4, 6}
+		c.Assert(utilTransferRegionCountParseLog(operator, content, expect), Equals, true)
```

Comment:
You support transfer leader, I think the name `TransferRegionCounter` is a bit unsuitable.

Reply:
Thanks